tt_metal/common/core_descriptor.hpp

```diff
@@ -69,14 +69,14 @@ inline const std::string get_product_name(tt::ARCH arch, uint32_t num_harvested_
 
 const core_descriptor_t &get_core_descriptor_config(chip_id_t device_id, const uint8_t num_hw_cqs, CoreType dispatch_core_type);
 
-inline uint32_t get_l1_bank_size(chip_id_t device_id, const uint8_t num_hw_cqs, CoreType dispatch_core_type) {
+inline std::optional<uint32_t> get_storage_core_bank_size(chip_id_t device_id, const uint8_t num_hw_cqs, CoreType dispatch_core_type) {
     const core_descriptor_t &core_desc = get_core_descriptor_config(device_id, num_hw_cqs, dispatch_core_type);
     const metal_SocDescriptor &soc_desc = tt::Cluster::instance().get_soc_desc(device_id);
-    uint32_t l1_bank_size = core_desc.storage_core_bank_size.has_value()
```

Comment:
this is only called during device initialization but there are a bunch of APIs in command_queue_interface.hpp that are on perf critical path. I left a comment there because currently the changes I made are uncached but Ill measure perf and cache them accordingly.